lib/membrane_rtc_engine/engine.ex

```diff
@@ -979,6 +931,21 @@ defmodule Membrane.RTC.Engine do
     endpoint_id = opts[:endpoint_id] || opts[:peer_id] || UUID.uuid4()
     endpoint_name = {:endpoint, endpoint_id}
 
+    with %Endpoint.WebRTC{} <- endpoint_entry do
+      metadata =
+        case state.peers do
+          %{^endpoint_id => %{metadata: metadata}} -> metadata
+          _else -> nil
+        end
```

Comment:
That's not needed. Peers are structs of type `Peer.t()` so they always have `metadata` field.